src/main/java/com/hedvig/contract/util/LegacyUtils.kt

```diff
@@ -69,6 +75,16 @@ class LegacyUtils(
     }
     return product
   }
+
+  fun getContractStateFromAgreementId(agreementId: UUID): ContractState<AgreementState> {
+    val optionalAgreementState = agreementStateRepository.findById(agreementId)
```

Comment:
There is a method called `findByIdOrNull` that is more "Kotlin"
```suggestion
    val optionalAgreementState = agreementStateRepository.findByIdOrNull(agreementId)
```

Reply:
This one is pretty neat 

```suggestion
    return agreementStateRepository.findById(agreementId)
      ?: throw ContractNotFoundException("No contract found with agreementId $agreementId")
```